### PR TITLE
Make tests that modify static variables run without parallelism

### PR DESCRIPTION
**Pull request by Shane32**
Fixes remainder of #2982 issues (hopefully).

See:
- https://xunit.net/docs/shared-context
- https://xunit.net/docs/running-tests-in-parallel

> By default, each test class is a unique test collection. Tests within the same test class will not run in parallel against each other.

> If we need to indicate that multiple test classes should not be run in parallel against one another, then we place them into the same test collection. This is simply a matter of decorating each test class with an attribute that places them into the same uniquely named test collection:

> Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

So with these changes,
  1. All tests that modify static variables are now placed in the same "test collection" called "StaticTests", so that all those tests "will not run in parallel against each other.
  2. The new test collection (called "StaticTests") has parallelism disabled, which means it will run sequentially after all other tests complete

So basically all tests within a class marked with `[Collection("StaticTests")]` are guaranteed to not run in parallel with any other tests.

All tests that use the `CultureTestHelper` or modify `GlobalSwitches` have had their test classes marked with `[Collection("StaticTests")]`

### CODE REVIEW

File: src/GraphQL.Tests/CultureTestHelper.cs [hunk 1]
```diff
@@ -14,6 +14,11 @@ public static class CultureTestHelper
             new CultureInfo(CultureInfo.CurrentCulture.Name)
         };
 
+        /// <summary>
+        /// Executes the specified delegate with a variety of cultures.
+        /// Be sure to mark the test class with <c>[Collection("StaticTests")]</c>
+        /// </summary>
+        /// <param name="scope"></param>
```
Comment (Shane32) [inline src/GraphQL.Tests/CultureTestHelper.cs:21]:
```suggestion
```

**Comment by sungam3r** [comment]
🤔 I think some time long ago I have already tried to do the same and I failed apparently because I missed point 2 about disable parallelization for dedicated collection so moving it after others.

File: src/GraphQL.Tests/CultureTestHelper.cs [hunk 2]
```diff
@@ -14,6 +14,10 @@ public static class CultureTestHelper
             new CultureInfo(CultureInfo.CurrentCulture.Name)
         };
 
+        /// <summary>
+        /// Executes the specified delegate with a variety of cultures.
+        /// Be sure to mark the test class with <c>[Collection("StaticTests")]</c>
+        /// </summary>
```
Comment (sungam3r) [inline src/GraphQL.Tests/CultureTestHelper.cs:20]:
```suggestion
        /// Be sure to mark the test class with <c>[Collection("StaticTests")]</c>
        /// to avoid the interference of static variables.
        /// </summary>
```

File: src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 namespace GraphQL.Tests.Types
 {
+    [Collection("StaticTests")]
```
Comment (sungam3r) [inline src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs:11]: Try to uncomment all lines after
> // disabled to make tests stable without touch GlobalSwitches

File: src/GraphQL.Tests/ObjectExtensionsTest.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 namespace GraphQL.Tests
 {
+    [Collection("StaticTests")]
```
Comment (sungam3r) [inline src/GraphQL.Tests/ObjectExtensionsTest.cs:3]: As I understand, there is no need to capture all the tests from test classes to make them sequential - only ones that uses `UseCultures` but I'm OK to keep as is since performance drop is almost invisible.

File: src/GraphQL.Tests/StaticTestsCollection.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+namespace GraphQL.Tests;
+
+// decorate any test classes containing tests that modify global variables with
+//   [Collection("StaticTests")]
+// these tests will run sequentially after all other tests are complete
+// be sure to restore the global variables to their initial state with a finally block
+
+[CollectionDefinition("StaticTests", DisableParallelization = true)]
+public class StaticTestsCollection
```
Comment (sungam3r) [inline src/GraphQL.Tests/StaticTestsCollection.cs:9]: Class name can be arbitrary?

**Review by sungam3r — APPROVED** [review]

File: src/GraphQL.Tests/ObjectExtensionsTest.cs (hunk 4 above)
Comment (Shane32) [reply src/GraphQL.Tests/ObjectExtensionsTest.cs:3]: That’s what I was thinking as well

File: src/GraphQL.Tests/StaticTestsCollection.cs (hunk 5 above)
Comment (Shane32) [reply src/GraphQL.Tests/StaticTestsCollection.cs:9]: Yes

File: src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs (hunk 3 above)
Comment (Shane32) [reply src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs:11]: Done
